.github/workflows/pythonpackage.yml

```diff
@@ -195,7 +202,7 @@ jobs:
   upload_pypi:
     needs: [ build_wheels, build_sdist, Metadata ]
     # upload to PyPI on every tag
-    if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags/')
+    if: github.event_name == 'push' && github.ref_type == 'tag'
```

Comment:
simpler filter with the same result
Unfortunately, selection of target from tag name/release type (alpha/beta/rc) is not so simple task and require tag name parse with external actions (cryptography publish not automatically by tags, but by manual trigger)